products.js

```diff
@@ -39,31 +45,93 @@ function tunedImageFnodes(nodeToCssMap, coeffImgSize = 2.5, coeffImgHasSrc = 2.5
+            (fnode.element.getAttribute(attrs[i]).includes(title) || title.includes(fnode.element.getAttribute(attrs[i])))){
+              return 100 * coeffImgTitle;
+            }
       }
+      return 1;
+    }
 
-      if (fnode.element.getAttribute('title') &&
-         (fnode.element.getAttribute('title').includes(title) || title.includes(fnode.element.getAttribute('title'))) ||
-          fnode.element.getAttribute('alt') &&
-         (fnode.element.getAttribute('alt').includes(title) || title.includes(fnode.element.getAttribute('alt')))) {
-               return 100 * coeffImgTitle;
+    function itemprop(fnode){
+      if(fnode.element.hasAttribute('itemprop') && fnode.element.getAttribute('itemprop').match(/(image|main|product|hero|feature)/i)){
+        return 1000 * coeffItemprop;
       }
-      return 1;
     }
 
+
     function keywords(fnode) {
       if(fnode.element.hasAttribute('src') && fnode.element.src.match(/(thumb|logo|icon)/i)){
-        return 0.5;
-      } else if (fnode.element.hasAttribute('src') && fnode.element.src.match(/(hero|main|product|feature)/i) ||
-                 fnode.element.id.match(/(hero|main|product|large|feature)/i) ||
-                 fnode.element.classList[0] && fnode.element.classList[0].match(/(hero|main|product|primary|feature)/i) ||
-                 fnode.element.hasAttribute('itemprop') && fnode.element.getAttribute('itemprop').match(/(image|main|product|hero|feature)/i)){
-        return 1000 * coeffKeywords;
+        return 0.5 * coeffBadKeywords;
+      }
+
+      const attrs = ['src', 'id', 'alt', 'title'];
+      for(let i = 0; i < attrs.length; i++){
+        if(fnode.element.hasAttribute(attrs[i]) && fnode.element.getAttribute(attrs[i]).match(/(main|product|hero|feature|primary)/i)){
+          return 1000 * coeffGoodKeywords;
+        }
+      }
+
+      for(let i = 0; i < fnode.element.classList.length; i++){
+        if(fnode.element.classList[i].match(/(hero|main|product|primary|feature)/i)){
+          return 1000 * coeffClassKeywords;
+        }
       }
       return 1;
     }
 
+    function titlewords(fnode){
+      const titleWords = title.replace('|', '').split(' ');
+      const attrs = ['src', 'title', 'alt'];
+      let start = 1;
+      for(let j = 0; j < attrs.length; j++){
+        for(let i = 0; i < titleWords.length; i++){
+            if(titleWords[i].match(/^[a-z0-9]+$/i)){
+              const regex = new RegExp(titleWords[i], "i");
+              if(fnode.element.hasAttribute(attrs[j]) && fnode.element.getAttribute(attrs[j]).match(regex)){
+                start+= fnode.element.getAttribute(attrs[j]).match(regex).length;
+              }
+          }
+        }
+      }
+      //exaggerating the differences a little
+      return Math.pow(titlewordsBase, start) * coefftitlewords;
+    }
+
+    function aboveTheFold(fnode){
+      const css = nodeToCssMap.get(fnode.element);
+      if(css.top < VIEWPORT_HEIGHT){
```

Comment:
I don't think so. getBoundingClientRect() gives distance from the top of window, which starts at 0.